components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/throttling/ThrottleHandler.java

```diff
@@ -380,36 +385,39 @@ private boolean doRoleBasedAccessThrottlingWithCEP(MessageContext synCtx, Config
+                                            applicationLevelThrottleKey, resourceLevelThrottleKey, apiLevelThrottleKey,
+                                            subscriptionLevelThrottleKey, apiContext, apiVersion, subscriberTenantDomain,
+                                            apiTenantDomain, applicationId, getThrottleDataHolder().getKeyTemplateMap(),
+                                            synCtx)) {
+                                        //Pass message context and continue to avoid performance issue.
+                                        //Did not throttled at any level. So let message go and publish event.
+                                        //publish event to Global Policy Server
+                                        if (isHardLimitThrottled(synCtx, authContext, apiContext, apiVersion)) {
+                                            isThrottled = true;
 
+                                        } else {
+                                            throttleDataPublisher.publishNonThrottledEvent(applicationLevelThrottleKey,
+                                                    applicationLevelTier, apiLevelThrottleKey, apiLevelTier,
+                                                    subscriptionLevelThrottleKey, subscriptionLevelTier,
+                                                    resourceLevelThrottleKey, resourceLevelTier, authorizedUser, apiContext,
+                                                    apiVersion, subscriberTenantDomain, apiTenantDomain, applicationId,
+                                                    synCtx, authContext);
+                                        }
                                     } else {
-                                        throttleDataPublisher.publishNonThrottledEvent(applicationLevelThrottleKey,
-                                                applicationLevelTier, apiLevelThrottleKey, apiLevelTier,
-                                                subscriptionLevelThrottleKey, subscriptionLevelTier,
-                                                resourceLevelThrottleKey, resourceLevelTier, authorizedUser, apiContext,
-                                                apiVersion, subscriberTenantDomain, apiTenantDomain, applicationId,
-                                                synCtx, authContext);
-                                    }
-                                } else {
-                                    if (log.isDebugEnabled()) {
-                                        log.debug("Request throttled at custom throttling");
-                                    }
-                                    synCtx.setProperty(APIThrottleConstants.THROTTLED_OUT_REASON,
-                                            APIThrottleConstants.CUSTOM_POLICY_LIMIT_EXCEED);
-                                    isThrottled = true;
+                                        if (log.isDebugEnabled()) {
```

Comment:
This check is not needed since no concats

Reply:
Fixed with 2a6fbdd3ac9ce54ca7bc94a45328a41826836a80